core-test/src/test/scala/org/bitcoins/core/protocol/tlv/EventDescriptorTest.scala

```diff
@@ -0,0 +1,241 @@
+                                count = UInt32(20),
+                                step = UInt16(5),
+                                unit = "",
+                                precision = Int32.negOne)
+    val range =
+      NumericRange[BigDecimal](start = -1.0,
+                               end = 9.0,
+                               step =
+                                 rangeEventBasePrecision1.step.toInt * 0.1)(
+        BigDecimalAsIfIntegral).toVector
+    assert(rangeEventBasePrecision1.stepDecimal == 0.5)
+    assert(rangeEventBasePrecision1.min == -1)
+    assert(rangeEventBasePrecision1.max == 8.5)
+    assert(rangeEventBasePrecision1.outcomesBigDec == range)
+
+    assert(rangeEventBasePrecision1.outcomes == range.map(_.toString()))
+  }
+
+  it must "be illegal to have num digits be negative or zero" in {
+    intercept[IllegalArgumentException] {
+      UnsignedDigitDecompositionEventDescriptor(base = UInt16(10),
+                                                numDigits = UInt16.zero,
+                                                unit = "BTC/USD",
+                                                precision = Int32.zero)
+    }
+    intercept[IllegalArgumentException] {
+      SignedDigitDecompositionEventDescriptor(base = UInt16(10),
+                                              numDigits = UInt16.zero,
+                                              unit = "",
+                                              precision = Int32.zero)
+    }
+  }
+
+  it must "serialize and deserialize a range event" in {
+    val hex = "fdd80815fffffffe0000000400010642544355534400000000"
+    val re = RangeEventDescriptorV0TLV(start = Int32(-2),
+                                       count = UInt32(4),
+                                       step = UInt16.one,
+                                       unit = "BTCUSD",
+                                       precision = Int32.zero)
+
+    val reFromHex = RangeEventDescriptorV0TLV.fromHex(hex)
+    assert(reFromHex == re)
+    assert(re.hex == hex)
+  }
+
+  it must "create a unsigned digit decomposition event" in {
+    val descriptor =
+      UnsignedDigitDecompositionEventDescriptor(base = UInt16(10),
+                                                numDigits = UInt16(1),
+                                                unit = "BTC/USD",
+                                                precision = Int32.zero)
+
+    assert(descriptor.max == 9)
+    assert(descriptor.min == 0)
+    val range = 0.until(10).toVector
+    assert(descriptor.outcomesBigDec == range)
+    assert(descriptor.outcomes == range.map(i => i.toString))
+
+    val descriptor1 = descriptor.copy(numDigits = UInt16(2))
+    assert(descriptor1.max == 99)
+    assert(descriptor1.min == 0)
+    val expected1 = 0.until(100)
+    val expectedString1 = expected1.map { i =>
+      if (i < 10) s"0${i}"
+      else i.toString
+    }
+    assert(descriptor1.outcomes == expectedString1)
```

Comment:
if outcomes is what the oracle signs then this should be `Vector(Vector("0", "0"), Vector("0", "1"), ...`

Reply:
same below